### PR TITLE
chore: [FX-2570] Split contribution workflow into reusable parts

### PR DESCRIPTION
**Pull request by LashaJini**
[FX-2570](https://toptal-core.atlassian.net/browse/FX-2570?atlOrigin=eyJpIjoiNjUyNTQyMDk5MjdlNDUzZmEyODQ0ZDg3NDk1ZmVlOWEiLCJwIjoiaiJ9)

### Description

Use `notify-jira-about-contribution` GH action which already does what we want:
- handle draft PRs
- ignore dependabot PRs

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| Insert screenshots or screen recordings | Insert screenshots or screen recordings |

### Review

- [ ] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [ ] Annotate all `props` in component with documentation
- [ ] Create `examples` for component
- [ ] Ensure that deployed demo has expected results and good examples
- [ ] Ensure that tests pass by running `yarn test`
- [ ] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [ ] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

### CODE REVIEW

File: .github/workflows/handle-contribution.yml [hunk 1]
```diff
@@ -1,40 +1,28 @@
...
-              :pray: please, help us improve, [rate your contributing experience](https://forms.gle/unzJBEJAHRz7LEtX8) after merge
-              `
-            })
+          repository: toptal/davinci-github-actions
+          token: ${{ secrets.TOPTAL_DEVBOT_TOKEN }}
+          ref: 'gha/notify-jira-about-contribution'
+          path: ./.github/actions/
+
+      - uses: ./.github/actions/notify-jira-about-contribution
+        if: github.event.pull_request.draft == false && !(github.actor == 'dependabot[bot]')
```
Comment (denieler) [inline .github/workflows/handle-contribution.yml:22]: shouldn't it be?

```suggestion
        if: github.event.pull_request.draft == false && !(github.actor == 'dependabot[bot]')
```
```suggestion
        if: github.event.pull_request.draft !== false && github.actor !== 'dependabot[bot] && ${{!contains(fromJson('["denieler", "TomasSlama", "OleksandrNechai", "OndrejTuma", "dmaklygin", "LashaJini", "separatio", "augustobmoura", "ozgurkececioglu", "toptal-build"]'), github.event.pull_request.user.login)}}'
```

Comment (LashaJini) [reply .github/workflows/handle-contribution.yml:22]: We can have `!==` in github actions? :smiley:

Comment (LashaJini) [reply .github/workflows/handle-contribution.yml:22]: The main issue is coming from something else, but yeah, it can be like this.

Comment (LashaJini) [reply .github/workflows/handle-contribution.yml:22]: We already have `is-team-member` reusable gha.

File: .github/workflows/handle-contribution.yml [hunk 2]
```diff
@@ -1,40 +1,26 @@
...
-              `
-            })
+          repository: toptal/davinci-github-actions
+          token: ${{ secrets.TOPTAL_DEVBOT_TOKEN }}
+          path: ./.github/actions/
+
+      - uses: ./.github/actions/notify-jira-about-contribution
+        with:
+          team: frontend-experience-eng
+          repo: picasso
```
Comment (separatio) [inline .github/workflows/handle-contribution.yml:23]: We don't only have contributions in Picasso.
```
repo: context.repo.repo,
```

Is this accounted for anywhere?

Comment (LashaJini) [reply .github/workflows/handle-contribution.yml:23]: I am not sure if I understand `context.repo.repo`. The `repo` parameter is used in our reusable GH action `notify-jira-about-contribution` in `davinci-github-actions` repo.

**Review by dmaklygin — APPROVED** [review]
Good job, the only thing with the name of the repo

File: .github/workflows/handle-contribution.yml (hunk 2 above)
Comment (LashaJini) [reply .github/workflows/handle-contribution.yml:23]: Sure I will do it :+1:

**Review by OndrejTuma — APPROVED** [review]
Awesome 🎉 👍

**Review by separatio — APPROVED** [review]
Code looks good. Didn't test. :+1:

**Pull request merged**